adhocracy4/maps_react/static/a4maps_react/Map.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useRef, useImperativeHandle } from 'react'
+import { MapContainer, GeoJSON } from 'react-leaflet'
+import MaplibreGlLayer from './MaplibreGlLayer'
+import ZoomControl from './ZoomControl'
+
+const polygonStyle = {
+  color: '#0076ae',
+  weight: 2,
+  opacity: 1,
+  fillOpacity: 0.2
+}
+
+const Map = React.forwardRef(function Map (
+  { attribution, baseUrl, polygon, omtToken, children, ...rest }, ref
+) {
+  const map = useRef()
+  // forwarding our map ref
+  useImperativeHandle(ref, () => map.current)
+  const refCallback = (polygon) => {
+    if (!map.current || !polygon) {
+      return
+    }
+    map.current.fitBounds(polygon.getBounds())
+    map.current.options.minZoom = map.current.getZoom()
```

Comment:
just to check: do we need the custom zoom controls because the zoom out button is not disabled on first load? if that's the only reason using `map.current.setMinZoom()` works for me here, so we wouldn't need a custom component just for that

Reply:
Huh, good catch! That's indeed so much better.